molecular/util/FileServer.h

```diff
@@ -99,6 +100,13 @@ class FileServer
 		}
 	}
 
+	void SetFileList(const std::string& files)
+	{
+		// This class does not support appending so it needs to be cleared.
+		mDirectoryContents = StringStore();
+		mDirectoryContents.LoadFromText(files.data(), files.size());
```

Comment:
I guess you could just add support for appending if that is desired.

Reply:
It's not desired, I was just explaining why it needs to be recreated. I've made that a little bit more clear now